core/src/main/java/org/eclipse/hono/util/HonoProtonHelper.java

```diff
@@ -308,4 +310,21 @@ public static void closeAndFree(
             link.free();
         });
     }
+
+    /**
+     * Sets a handler that will be invoked when the given message delivery of a received message gets updated
+     * from the remote peer before the local receiver updates the delivery.
+     * <p>
+     * A corresponding scenario would be that the remote sender of the message only waits a limited time for
+     * the disposition update and afterwards aborts (i.e. releases) the delivery from the sender side. With
+     * the handler here, the receiver can get notified of such a delivery update.
+     *
+     * @param delivery The delivery to set the handler on. Must be a delivery provided by a <em>ProtonReceiver</em> handler.
```

Comment:
IMHO it is worth mentioning that the delivery is actually expected to be of type `ProtonDeliveryImpl` ...